### PR TITLE
Add unit tests for model deck reference validation

### PR DESCRIPTION
**Pull request by lchoquel**
## Summary
- Add comprehensive test suite for validating model deck references (aliases, presets, waterfalls)
- Tests cover LLM, Extract, and ImgGen model types
- Validates that references point to valid targets and detects circular alias chains
- Prevents configuration errors from being discovered at runtime

## Test plan
- [x] Run `make test-xdist` to verify all new tests pass
- [ ] Review test coverage for edge cases in model deck configurations

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> Adds comprehensive validation and tests for model deck references and tightens type checks in inference resolution.
> 
> - New `tests/.../model_deck_validation_utils.py` helpers and parameterized suites validating `aliases`, `presets`, and `waterfalls` (detect unknown targets, wrong `ModelType`, empty waterfalls, and circular aliases)
> - Extend `tests/.../test_model_deck.py` with type-mismatch case and prefixed reference scenarios; new `test_model_deck_references.py` covers real blueprint/backends + gateway models
> - Model changes: `validate_inference_models` now uses each spec’s `model_type`; `get_optional_inference_model` skips and warns on `model_type` mismatch
> - Test housekeeping: replace class-level `@pytest.mark.asyncio(loop_scope="class")` with per-test `@pytest.mark.asyncio` across content render tests
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7a6b7996eccc7e68e7f76e3c35fc28c4a4dc2ce8. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: tests/unit/pipelex/cogt/models/test_model_deck_validation_helpers.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        unique_cycles: list[tuple[str, list[str]]] = []
+        seen_cycles: set[frozenset[str]] = set()
+
+        for start_alias, cycle_path in cycles:
+            cycle_set = frozenset(cycle_path)
+            if cycle_set not in seen_cycles:
+                seen_cycles.add(cycle_set)
+                unique_cycles.append((start_alias, cycle_path))
+
+        return unique_cycles
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/test_model_deck_validation_helpers.py]: Fixed

File: tests/unit/pipelex/cogt/models/test_model_deck_references.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+                        invalid_refs.append((alias_name, target_value, f"waterfall '{ref.name}' not found"))
+                case ModelReferenceKind.PRESET:
+                    # Alias should NOT point to a preset
+                    invalid_refs.append((alias_name, target_value, "aliases cannot reference presets ($ prefix)"))
+                case ModelReferenceKind.HANDLE:
+                    # Validate direct handle exists in known models
+                    if ref.name not in known_model_handles:
+                        invalid_refs.append((alias_name, target_value, f"model handle '{ref.name}' not found in backends"))
+
+        return invalid_refs
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/test_model_deck_references.py]: Tentatively fixed, to be confirmed

File: tests/unit/pipelex/cogt/models/test_model_deck_references.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+                case ModelReferenceKind.PRESET:
+                    invalid_refs.append((preset_name, model_value, "presets cannot reference other presets ($ prefix)"))
+                case ModelReferenceKind.HANDLE:
+                    if ref.name not in known_model_handles:
+                        invalid_refs.append((preset_name, model_value, f"model handle '{ref.name}' not found in backends"))
+                    elif known_model_handles[ref.name] != expected_model_type:
+                        actual_type = known_model_handles[ref.name]
+                        invalid_refs.append(
+                            (preset_name, model_value, f"model handle '{ref.name}' has type '{actual_type}' but expected '{expected_model_type}'")
+                        )
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/test_model_deck_references.py:157]: Fixed

File: tests/unit/pipelex/cogt/models/model_deck_validation_utils.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                break
+
+            visited.append(current)
+            target = aliases[current]
+            ref = ModelReference.parse(target)
+
+            if ref.kind == ModelReferenceKind.ALIAS:
+                current = ref.name
+            else:
+                break
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/model_deck_validation_utils.py]: Fixed

File: tests/unit/pipelex/cogt/models/model_deck_validation_utils.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            case ModelReferenceKind.PRESET:
+                invalid_refs.append((alias_name, target_value, "aliases cannot reference presets ($ prefix)"))
+            case ModelReferenceKind.HANDLE:
+                if ref.name not in known_model_handles:
+                    invalid_refs.append((alias_name, target_value, f"model handle '{ref.name}' not found in backends"))
+                elif known_model_handles[ref.name] != expected_model_type:
+                    actual_type = known_model_handles[ref.name]
+                    invalid_refs.append(
+                        (alias_name, target_value, f"model handle '{ref.name}' has type '{actual_type}' but expected '{expected_model_type}'")
+                    )
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/model_deck_validation_utils.py:51]: Fixed

File: tests/unit/pipelex/cogt/models/test_model_deck_references.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+        img_gen_deck = model_deck_blueprint.img_gen
+
+        cycles = find_circular_aliases(aliases=img_gen_deck.aliases)
+
+        if cycles:
+            error_lines = ["Circular ImgGen alias references detected:"]
+            for _start_alias, cycle_path in cycles:
+                cycle_str = " -> ".join(cycle_path)
+                error_lines.append(f"  - {cycle_str}")
+            pytest.fail("\n".join(error_lines))
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/test_model_deck_references.py:298]: Fixed

File: tests/unit/pipelex/cogt/models/model_deck_validation_utils.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        known_model_handles: Mapping of model handle -> ModelType
+        expected_model_type: The expected model type for this deck (LLM, TEXT_EXTRACTOR, IMG_GEN)
+
+    Returns:
+        List of (alias_name, target_value, reason) tuples for invalid references
+    """
+    invalid_refs: list[tuple[str, str, str]] = []
+
+    for alias_name, target_value in aliases.items():
+        ref = ModelReference.parse(target_value)
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cogt/models/model_deck_validation_utils.py:33]: False positive
